onnx/defs/tensor/defs.cc

```diff
@@ -1794,4 +1794,80 @@ ONNX_OPERATOR_SET_SCHEMA(
 
           propagateShapeFromInputToOutput(ctx, 0, 0);
         }));
+
+static const char* GatherND_ver11_doc = R"DOC(
+Given `data` tensor of rank r >= 1, and `indices` tensor of rank q >= 1, gathers
+slices of `data` into an output tensor of rank q + r - indices_shape[-1] - 1.
+Example 1:
+  data    = [[0,1],[2,3]]
+  indices = [[0,0],[1,1]]
+  output  = [0,3]
+Example 2:
+  data    = [[0,1],[2,3]]
+  indices = [[1],[0]]
+  output  = [[2,3],[0,1]]
+Example 3:
+  data    = [[[0,1],[2,3]],[[4,5],[6,7]]]
+  indices = [[0,1],[1,0]]
+  output  = [[2,3],[4,5]]
+Example 4:
+  data    = [[[0,1],[2,3]],[[4,5],[6,7]]]
+  indices = [[[0,1]],[[1,0]]]
+  output  = [[[2,3]],[[4,5]]]
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    GatherND,
+    11,
+    OpSchema()
+    .SetDoc(GatherND_ver11_doc)
+    .Input(0, "data", "Tensor of rank r >= 1.", "T")
+    .Input(1, "indices", "Tensor of rank q >= 1.", "tensor(int64)")
+    .Output(0, "output", "Tensor of rank q + r - indices_shape[-1] - 1.", "T")
+    .TypeConstraint(
+        "T",
+        OpSchema::all_tensor_types(),
+        "Constrain input and output types to any tensor type.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+  // Type inference
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // Shape inference
+  if (!hasNInputShapes(ctx, 2)) {
+    // cannot proceed with shape or rank inference
+    return;
+  }
+
+  const auto& data_shape = ctx.getInputType(0)->tensor_type().shape();
+  const auto data_rank = data_shape.dim_size();
+
+  const auto& indices_shape = ctx.getInputType(1)->tensor_type().shape();
+  const auto indices_rank = indices_shape.dim_size();
+
+  if (data_rank < 1 || indices_rank < 1)
+    fail_shape_inference(
+        "Both `data` and `indices` tensors need to have rank larger than 0.");
+
+  // cannot ascertain if the input shapes are valid if shape of `indices` 
+  // is missing last dimension value so return at this point
+  if (!indices_shape.dim(indices_rank - 1).has_dim_value())
+    return;
+
+  const auto last_indice_dimension =
```

Comment:
Some people might see the name `last_indice_dimension` as a typo. Maybe rename this to `last_index`dimension`

Reply:
Fixed